samples/ipc/ipc_service/src/main.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+
+#include <ipc/ipc_service.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(host, LOG_LEVEL_INF);
+
+#define MLEN_0 40
+
+struct data_packet {
+	unsigned char message;
+	unsigned char data[100];
+};
+
+
+K_SEM_DEFINE(bound_sem, 0, 1);
+
+static void ep_bound(void *priv)
+{
+	k_sem_give(&bound_sem);
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	char received_data = *((char *)data);
+	static char expected_message = 'a';
+	static uint16_t expected_len = MLEN_0;
+
+	static unsigned long long cnt;
+	static unsigned int stats_every;
+	static uint32_t start;
+	static unsigned int err;
+
+	if (start == 0) {
+		start = k_uptime_get_32();
+	}
+
+	if (received_data != expected_message) {
+		err++;
+	}
+
+	if (len != expected_len) {
+		err++;
+	}
+
+	expected_message++;
+	expected_len++;
+
+	cnt += len;
+
+	if (expected_message > 'z') {
```

Comment:
if you convert from letters to number this could be simplified :)

Reply:
at line 45, in_count != our_count
at line 53 out_count++
get rid of line 58
at line 59 if (out_count % 1024 == 0) // or 2048 or something else

Reply:
There is no simplification at all, but okay i will convert to numbers.